### DISCUSSION TITLE
More reasonable alternative to forking?

### DISCUSSION THREAD

**Discussion by PavanLuca** [community/community#130288] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

I don't like having to fork a repository just to correct some typos. Since I can only do so (because I don't know how to code) to fix them I have to fork the whole repository, sometimes just to fix a few, or literally 1. In these cases it would be more reasonable to fork a single file, fix it, and make a pull request since the changes are really very limited.

**Comment by asdacosta** [comment]
Imagine the hassle if you had to navigate and fork different files if you wanted to make a lot of changes too. That will look messy and each file you fork must have a repo, so if you fork a single file from a repo then where will that file be placed in your repositories? GitHub set this convention to help in seamless workflow.

**Reply by PavanLuca** [reply]
In my idea each file that the user edits would be listed under a title that's the repository containing them and the user can manage them easily. There the user can view their details like the creation date, last edit date and applying actions (like changing notification option, delete the file and send it). The repository owner would receive a notification of the pull request that tells **(Username) sent you a pull request**, the quick options will be `View`, `Mark as read` and `Notify me later`. Repository owners will see a title per repository that they own with all notifications of pull requests, but if no pull request exists for a repository of theirs the page will contain no title (so there would be a screen like **No pull requests for your repositories for now. Maybe later…** 😄) while it will list titles for repositories that have at least 1 pull request. Tell me if it wouldn't be better for a repository owner, I don't have any so I don't know how a repository owner handles pull requests matters 😅.

**Comment by Yewzir** [comment]
@PavanLuca 

Why don't you simply edit a file in place, commit your changes and open a PR?

![image](https://github.com/community/community/assets/116750824/abc2cdf3-d71a-4fca-a221-2f7f05a8210b)

It'll create a new branch automatically, which is easy to delete afterwards. Very convenient for fixing typos or minor discrepancies in general.

**Reply by TheGiraffe3** [reply]
But you still have the fork.

**Reply by PavanLuca** [reply]
> Why don't you simply edit a file in place, commit your changes and open a PR?

Yes, I should try again doing so, I did once and stopped immediately after when I saw that Firefox or GitHub were asking me a confirmation before leaving the page or saving the changes. Also, after choosing either to save them or abort the changes I made weren't saved in any case (viewing the same file even after reloading the page didn't show my changes on the file preview) and the page was stopping in an apparently endless loading (the blue bar on the top of the page progressing really slow) so instead of trying again at that moment I just saved the file and made the changes offline for then uploading it.

**Reply by TheGiraffe3** [reply]
You have to say you're okay with discarding your unsaved changes, and it should work.

**Comment by spenserblack** [comment]
That's not really how Git works. With Git, all files' histories are tracked together as one history in the repository. The repository is *distributed,* meaning that each person that clones/forks the repository has *their own copy* of the history. When you fork, you create a history that *you* own. Then with a pull request you're saying "please pull the latest changes from *my* repository's history to *your* repository's history."

The only way I could see GitHub implementing a "single-file fork" would be by "pretending" to fork one file but actually forking the whole repo (since you need that history).

I suppose something similar to what you're asking could also be done by creating an submitting patch files (basically a file that defines changes and a commit message to apply). But patches have often been considered confusing, and I kind of doubt that the GitHub team would be willing to officially support them.

A bit off-topic, but I believe the usage that you're describing would be possible with a *centralized* version control system, where there's only one centralized history and users submit *files,* not *changes.*
